prettyprinter/src/Data/Text/Prettyprint/Doc/Render/Util/StackMachine.hs

```diff
@@ -72,11 +72,11 @@ renderSimplyDecorated text push pop = go []
     go _           SFail               = panicUncaughtFail
     go []          SEmpty              = mempty
     go (_:_)       SEmpty              = panicInputNotFullyConsumed
-    go stack       (SChar c rest)      = text (T.singleton c) <> go stack rest
-    go stack       (SText _l t rest)   = text t <> go stack rest
-    go stack       (SLine i rest)      = text (T.singleton '\n') <> text (T.replicate i " ") <> go stack rest
-    go stack       (SAnnPush ann rest) = push ann <> go (ann : stack) rest
-    go (ann:stack) (SAnnPop rest)      = pop ann <> go stack rest
+    go stack       (SChar c rest)      = text (T.singleton c) `mappend` go stack rest
```

Comment:
Was `<>` not part of `Data.Monoid` back then? If not, I would prefer a conditional local definition,

```haskell
#if !(DIAMOND_OPERATOR_SUPPORTED)
(<>) = mappend
#endif
```

Reply:
that `<>` would clash with semigroup's `<>`.